### DISCUSSION TITLE
Folder templates template for quick switcher

### DISCUSSION THREAD

**Discussion by Zachatoo** [SilentVoid13/Templater#955] [Templates Showcase]
This setup simulates the [folder templates](https://silentvoid13.github.io/Templater/settings.html#folder-templates) feature but in a way that allows me to type the name of the note without the folder name in the quick switcher, and then be prompted with the folder using `tp.system.suggester`.

For example, I can hit `CMD/CTRL O` to open the quick switcher, type the name of the note in the quick switcher dialog, hit `SHIFT ENTER` to create the note, and be prompted with which folder I want the note to go in and have that note's folder template be applied.

Full post with explanation: https://zachyoung.dev/posts/folder-templates-with-quick-switcher

```js
<%*
// Fallback template if no type is selected
let template = 'default-template';
// Get list of folder templates (setup in Templater settings)
const folderTemplates = app.plugins.plugins['templater-obsidian'].settings.folder_templates;
// Get default path for new files
const defaultNewFilePath = app.vault.getConfig("newFileFolderPath");
// If note is in default note location (likely from quick switcher)
if (tp.file.folder() === defaultNewFilePath) {
  // Prompt for folder template
  const selectedTemplate = await tp.system.suggester(folderTemplate => folderTemplate.folder, folderTemplates);
  // If folder template is selected and is not the picker template,
  const pickerTemplateFolder = '/';
  if (selectedTemplate && selectedTemplate.folder !== pickerTemplateFolder) {
    // Move file to folder template and set template to apply
    template = selectedTemplate.template;
    await tp.file.move(`${selectedTemplate.folder}/${tp.file.title}`);
  }
}
// Apply template
tR += await tp.file.include(`[[${template}]]`)
_%>
```

**Comment by danielo515** [comment]
I'm very surprised this doesn't have any comment. My main problem with quick open os that I have to type the full path to create a new file. This script simplifies that allowing me to focus on switching or creating the new note. This os great

**Reply by Zachatoo** [reply]
That reminds me, I've updated this a fair bit to use your configured folder template and default new note location settings, much more concise. I've updated this post now!

**Comment by danielo515** [comment]
Hey, sorry for complaining again about your nice template 😅 
I noticed that `app.vault.getConfig("newFileFolderPath");`  gives you a full path (in my case, nested) while `tp.file.folder()`, even if you set relative to false, gives you just the folder name. For that reason, if the folder is nested the comparison with the default folder location will not match. 
Not sure if it is better to make a more loose comparison (for example, ends with) or to just use a different API to get the destination folder

**Reply by danielo515** [reply]
Ok, forget what I said, to get the full path of the destination you need to set relative to true, which for me is absolutely contrary to intuitive

**Reply by Zachatoo** [reply]
100% agree, the docs should say `absolute` not `relative`

**Comment by danielo515** [comment]
@Zachatoo in the latest version of templater I'm having a problem with this. The template is triggering twice. I think this was a jug in templater reintroduced with the latest update

**Reply by Zachatoo** [reply]
Commented on your bug report here #1265 😄
